packages/gatsby/src/redux/reducers/components.ts

```diff
@@ -1,12 +1,40 @@
-const normalize = require(`normalize-path`)
-const { interpret } = require(`xstate`)
+import normalize from "normalize-path"
+import { interpret } from "xstate"
 
 import { componentMachine } from "../machines/page-component"
+import {
+  IGatsbyState,
+  IDeleteCacheAction,
+  ISetProgramStatusAction,
+  ICreatePageAction,
+  IQueryExtractedAction,
+  IQueryExtractedBabelSuccessAction,
+  IQueryExtractionBabelErrorAction,
+  IQueryExtractionGraphQLErrorAction,
+  IPageQueryRunAction,
+  IDeletePageAction,
+  IRemoveTemplateComponentAction,
+} from "../types"
 
 const services = new Map()
 let programStatus = `BOOTSTRAPPING`
 
-module.exports = (state = new Map(), action) => {
+type Actions =
+  | IDeleteCacheAction
+  | ISetProgramStatusAction
+  | ICreatePageAction
+  | IQueryExtractedAction
+  | IQueryExtractedBabelSuccessAction
+  | IQueryExtractionBabelErrorAction
+  | IQueryExtractionGraphQLErrorAction
+  | IPageQueryRunAction
+  | IRemoveTemplateComponentAction
+  | IDeletePageAction
```

Comment:
Could we use the `ActionsUnion` from `../types` here instead of creating a new one?
It could reduce the number of imports in this file.

Reply:
Second that. We actually should do it because all actions are dispatched to reducers even if they are not handled in them (that's why reducers do have default cases or fallback that return unchanged `state` if it didn't match `action.type`)

Reply:
Good idea, changing!